prio_processor/spark/udf.py

```diff
@@ -9,6 +9,28 @@
 logger = logging.getLogger(__name__)
 
 
+def encode_single(
+    batch_id: str,
+    n_data: int,
+    public_key_hex_internal: bytearray,
+    public_key_hex_external: bytearray,
+    payload: list,
+):
+    libprio.Prio_init()
+    public_key_internal, public_key_external = import_public_keys(
+        bytes(public_key_hex_internal), bytes(public_key_hex_external)
+    )
+    config = libprio.PrioConfig_new(
+        n_data, public_key_internal, public_key_external, batch_id.encode()
+    )
+    try:
+        a, b = libprio.PrioClient_encode(config, bytes(list(map(int, payload))))
+    except:
```

Comment:
Can you specify the set of exceptions more tightly? Blanket try/except can hide subtle bugs.

Reply:
I've added a the set that the exceptions should handle (RuntimeError for weird underlying libprio issues, ValueError/TypeError for null typees).